articles/101_build_tool.md

```diff
@@ -0,0 +1,213 @@
+
+### catkin_make
+
+`catkin_make` is provided by the ROS package `catkin` which contains the build system for ROS 1.
+It has been designed as the successor of `rosbuild` for ROS Fuerte.
+
+The tool invokes CMake only a single time and uses CMake's `include` function to process all packages in a single context.
+While this is an efficient approach since all targets across all packages can be parallelized it comes with significant disadvantages.
+Due to the single context all function names, targets and tests share a single namespace across all packages and on a larger scale this easily leads to collisions.
+The single context is also subject to side effects between the packages and sometimes requires adding additional target dependencies across package boundaries.
+
+### catkin_make_isolated
+
+`catkin_make_isolated` is provided by the ROS package `catkin` which contains the build system for ROS 1.
+It was developed after `catkin_make` to address the problems involved with building multiple packages in a single CMake context.
+
+The tool only supports CMake-based packages and builds each package in topological order using the command sequence common for CMake packages: `cmake`, `make`, `make install`.
+While each package can parallelize the build of its targets the packages are processed sequentially even if they are not (recursive) dependencies of each other.
+
+### ament_tools
+
+`ament_tools` is provided by a standalone Python 3 package used to build ROS 2 packages.
+It was developed to bootstrap the ROS 2 project, is therefore only targeting Python 3, and works on Linux, MacOS and Windows.
+In addition to CMake packages it also supports building Python packages and can infer meta information without requiring an explicit package manifest.
+The tool performs an "isolated" build like `catkin_make_isolated` (one CMake invocation per package) but also parallelizes the build of packages which have no (recursive) dependencies on each other.
+
+### catkin_tools
+
+[catkin_tools](https://catkin-tools.readthedocs.io/) is provided by a standalone Python package used to build ROS 1 packages.
+It was developed after `catkin_make` / `catkin_make_isolated` to build packages in parallel as well as provide significant usability improvements.
+The tool supports building CMake packages and builds them in isolation as well as supports parallelizing the process across packages.
+
+### Naming
+
+The existing build tools are all named by the build system they are supporting.
+In general it should be possible for a build tool to support multiple different build systems.
+Therefore a name for a build tool being derived from a single build system might mislead the users that the tool only works for that specific build system.
+To avoid confusion of the user it would be beneficial if the build tool would have a different name to avoid implying an undesired correlation.
```

Comment:
> as weak as "not going to happen"

Well, I don't see it that way 😃. A nice-to-have would be in favor of a "from scratch" tool or forking an existing tool to rename it, which might lead us to favor those options too. However, if it were a requirement it would categorically exclude a proposal to iterate on an existing tool without changing the name. I'm happy to have this as a nice-to-have as I've described here, but to not consider iterating on a tool without changing its name moving forward goes too far for me to agree right now.

I would need to see more discussion from more people about this for my mind to be changed.

> It is clearly not a requirement since any name would "work".

But according to this section, a name that is based on an existing build system would be bad, so not any name would work by that reasoning.

> so stating that this is what we want to do.

In that case, unfortunately 😞, I'm not sure I can agree with this. I'm certainly not completely opposed to it, but I wouldn't go so far as to say I'm confident that's the right direction to take. There are some possible negative impacts of having a new name. I think @sloretz pointed them out well enough that I don't have to reiterate them.

----

I'm still not clear on the distinction you're making. If a goal is a weaker version of a requirement, then how is that different than a nice-to-have? Maybe you could clarify: what's the difference (in your opinion) in a requirement and a goal and a nice-to-have?

Perhaps there's more precise terminology we can draw on, to avoid the nuances.